docs/storage.md

```diff
@@ -413,3 +409,519 @@ type: FILESYSTEM
+			},
+			{
+				"ulid": "01DMSFSXNE8Y76G5KCQ2BABYFA",
+				"minTime": 1568332800000,
+				"maxTime": 1568505600000
+			},
+			{
+				"ulid": "01DMYMM5SW0FPJSHQQQM05FBN9",
+				"minTime": 1568505600000,
+				"maxTime": 1568678400000
+			},
+			{
+				"ulid": "01DN3SDE1M9W1JG7JFSM5QFP2Y",
+				"minTime": 1568678400000,
+				"maxTime": 1568851200000
+			}
+		]
+	},
+	"version": 1,
+	"thanos": {
+		"labels": {
+            "cluster": "eu1",
+			"monitor": "prometheus",
+            "tenant": "team-a",
+			"replica": "1"
+		},
+		"downsample": {
+			"resolution": 0
+		},
+		"source": "compactor",
+       	"files": [
+       			{
+       				"rel_path": "index",
+       				"size_bytes": 1313
+       			}, // Trimmed items for readability.
+       	],
+        "version": 1
+	}
+}
+```
+
+Format in Go code can be found [here](../pkg/block/metadata/meta.go).
+
+##### External Labels
+
+External labels are the extremely important block metadata. They are stored in `meta.json` in `thanos.labels` section and allows
+to identify producer and owner of those blocks. This information will be used further by different Thanos components:
+
+* Those labels will be visible when data is queried. You can aggregate across those in PromQL etc.
+* [Querier](./components/query.md) to filter out store APIs to touch during query requests.
+* Many object storage readers like [compactor](./components/compact.md) and [store gateway](./components/store.md) which groups the blocks by external labels.
+This grouping allows horizontal scalability like sharding or concurrency.
+* Some of those labels can be chosen as **replication** labels. Querier and Compactor will then deduplicate such blocks identified by same HA groups.
+* Some of those labels can be chosen as **tenancy** labels. This allows read, write and storage isolation mechanism.
+
+The `meta.json` `thanos.labels` labels are filled during block upload / creation. For example:
+
+* Each produced TSDB block by Prometheus is labelled with Prometheus [external labels](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#configuration-file)
+by `sidecar` before upload to object storage.
+* Each produced TSDB block by `compact` is labelled with whatever source blocks had. The exception is deduplication process that removes chosen replica flag(s).
+* Each produced TSDB block by `receive` is labelled with labels given labels in repeated [receive](./components/receive.md) `--labels` flag.
+
+The recommended information that should be given in those labels:
+
+Example Prometheus useful external labels:
+
+* Replication information e.g `replica="0"`
+* Cluster, environment, zone, so target origin e.g `cluster="eu-1-production"` or `cluster="1",env="production",region="us-west1"`
+* Tenancy information e.g `tenant="organizationABC"`
+
+> NOTE: Be careful with receive external flags. Remote Write clients can stream any labels. If some label will duplicate with external label of receive, it will be masked
```

Comment:
```suggestion
> NOTE: Be careful with receive external flags. Remote Write clients can stream any labels. If some label will duplicate with the external label of receive, it will be masked
```